pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -267,6 +269,11 @@ public void shutDown() {
   /// If any error happened during the asynchronous execution, an error block will be sent to all receiver mailboxes.
   public CompletableFuture<Void> processQuery(WorkerMetadata workerMetadata, StagePlan stagePlan,
       Map<String, String> requestMetadata, @Nullable ThreadExecutionContext parentContext) {
+    String requestIdStr = Long.toString(QueryThreadContext.getRequestId());
+    long requestId = QueryThreadContext.getRequestId();
+
+    _resourceUsageAccountant.registerQueryCancelCallback(requestIdStr, () -> _opChainScheduler.cancel(requestId));
```

Comment:
Not super important, but we need to allocate a new instance per query here.
 
Instead we could change the signature of the QueryCancelCallback to receive the request id as argument and then we could create a single instance callback and reuse it.